pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -1965,6 +2023,7 @@ pj_bool_t pjsua_call_on_incoming(pjsip_rx_data *rdata)
             pjsua_perror(THIS_FILE, "Error initializing media channel", status);
             
             pjsip_dlg_inc_lock(dlg);
+            st_code = sip_err_code;
```

Comment:
When `pjsua_media_channel_init()` is done async-ly or returning `EPENDING`, e.g: due to waiting for TURN allocations, the library may reject the call if the media init fails, should we invoke the callback too?

Reply:
This is an interesting scenario indeed. And note that `on_incoming_call()` has been called for such case, whereas for other cases, it hasn't.

Reply:
Since the `on_incoming_call()` has been called, the new callback (`on_rejected_incoming_call()`) should not be called again. Another case would be if the called is answered with 200OK, however there's an issue (e.g: media init) and the call is rejected. The callback would not be called, however app still be informed from `on_call_state()`.

Reply:
Make sure the behavior is consistent on pjsua2 as well, since on pjsua2 onIncomingCall() is called earlier.

Reply:
As it is async, I guess there can be race between `on_incoming_call()` and `on_incoming_call_med_tp_complete()`, in other word, we cannot be sure whether `on_incoming_call()` has been called?